src/messaging/ExchangeContext.cpp

```diff
@@ -689,5 +695,21 @@ void ExchangeContext::ExchangeSessionHolder::GrabExpiredSession(const SessionHan
     GrabUnchecked(session);
 }
 
+bool ExchangeContext::ShouldPeerBeActive()
+{
+    // If we are not the initiator, it means the peer sent the first message.
+    // This means our peer is active if we are sending a message to them.
+    if (!IsInitiator())
+        return true;
+
+    // If we create an Ephemeral Exchange, it was just to generate a StandaloneAck
+    // Since we are sending an ack, we know the peer is active
```

Comment:
```suggestion
    // If we create an Ephemeral Exchange, it was just to generate a StandaloneAck.
    // Since we are sending an ack, we know the peer is active.
```